plugins/Observability/src/Observability.cpp

```diff
@@ -1850,10 +1838,41 @@ void Observability::showReport(bool b)
 	}
 }
 
+bool Observability::isMoon(QList<StelObjectP> & objectSelection)
+{
+   return !objectSelection.empty() && "Moon" == objectSelection[0]->getEnglishName();
+}
+
+bool Observability::isSun(QList<StelObjectP> & objectSelection)
+{
+   return !objectSelection.empty() && "Sun" == objectSelection[0]->getEnglishName();
+}
+
+bool Observability::isNotSSO(QList<StelObjectP> & objectSelection)
+{
+   if (objectSelection.empty()) {
+       return true; // no object selected, so the selection is the screen/space.
+   }
+   // This works because the Planet class only exists for in system objectSelection (including moon and sun).
+   Planet * planet = dynamic_cast<Planet *>(objectSelection[0].data());
+   return planet == Q_NULLPTR;
+}
+
+bool Observability::isSystemPlanet(QList<StelObjectP> & objectSelection)
```

Comment:
I am afraid this is not what the original intent was: "SSO but neither sun nor moon." Apart from planets, there are moons, asteroids, comets, KBOs, ...
You should probably just test the Planet name for "Sun" or "Moon" like before.

Reply:
On line 1868, it checks that the planet's type is "planet." 

As far as I can tell, the only planet objects with type=planet are the planets within the solar system, e.g., https://github.com/Stellarium/stellarium/blob/51db4ee66f176d5c94f6ebe95eb0a72d33dc9306/data/ssystem_major.ini#L451.

Doesn't checking that the `Planet` type is "planet" ensure that it's not a moon, asteroid, etc.?

Reply:
Yes, but that's the problem. How to check observability for e.g. Vesta? ("moon" in this context is only "The Moon" (Earth's moon))

Reply:
Ah, I see. I'll change this.